tests/components/igloohome/conftest.py

```diff
@@ -0,0 +1,73 @@
+"""Common fixtures for the igloohome tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, patch
+
+from igloohome_api import GetDeviceInfoResponse, GetDevicesResponse
+import pytest
+
+from homeassistant.components.igloohome.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+GET_DEVICE_INFO_RESPONSE_LOCK = GetDeviceInfoResponse(
+    id="123456",
+    type="Lock",
+    deviceId="OE1X123cbb11",
+    deviceName="Front Door",
+    pairedAt="2024-11-09T11:19:25+00:00",
+    homeId=[],
+    linkedDevices=[],
+    batteryLevel=100,
+)
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.igloohome.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def auth_successful():
+    """Set up the Auth module to always successfully operate."""
+    return patch(
+        "igloohome_api.Auth.async_get_access_token",
+        return_value="mock_access_token",
+    )
```

Comment:
```suggestion
@pytest.fixture
async def mock_auth() -> Generator[MagicMock]:
    """Mock the auth class."""
    with (
            patch(
            "igloohome_api.Auth.",
            autospec=True
        ) as mock_auth,
        patch(
            "igloohome_api.config_flow.Auth",
            new=mock_auth,
        )
    ):
        auth = mock_auth.return_value
        auth.async_get_access_token.return_value = True
        yield auth
```
I'd mock away the entire object. This should also solve your problem below you were asking about

Reply:
Hey @zweckj , I've tried integrating your suggested changes here but ran into an issue that I'm trying to figure out. Do you what the issue is here?

(Needed to switch the patch targets to the `IgloohomeAuth` objects being constructed in the `igloohome` component modules)

```
FAILED                                    [100%]INFO:homeassistant.loader:Loaded igloohome from homeassistant.components.igloohome

tests/components/igloohome/test_config_flow.py:20 (test_form)
hass = <HomeAssistant RUNNING>
mock_setup_entry = <AsyncMock name='async_setup_entry' id='281473424716768'>
mock_auth = <NonCallableMagicMock name='IgloohomeAuth()' spec='Auth' id='281473422491520'>

    async def test_form(
        hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_auth
    ) -> None:
        """Test we get the form."""
        result = await hass.config_entries.flow.async_init(
            DOMAIN, context={"source": config_entries.SOURCE_USER}
        )
        assert result["type"] == FlowResultType.FORM
        assert result["errors"] == {}
    
>       with mock_auth:
E       TypeError: 'NonCallableMagicMock' object does not support the context manager protocol

test_config_flow.py:31: TypeError

Process finished with exit code 1
```

Reply:
you don't use that with `with`, but just add it to the parameters of your test

Reply:
Thanks for the hint.